libs/server/Storage/Session/ObjectStore/SortedSetOps.cs

```diff
@@ -886,6 +886,155 @@ public GarnetStatus SortedSetScan<TObjectContext>(byte[] key, ref ObjectInput in
+
+                if (status == GarnetStatus.WRONGTYPE)
+                {
+                    return GarnetStatus.WRONGTYPE;
+                }
+
+                count = pairs?.Count ?? 0;
+
+                if (count > 0)
+                {
+                    SortedSetObject newSetObject = new();
+                    foreach (var (element, score) in pairs)
+                    {
+                        newSetObject.Add(element, score);
+                    }
+                    _ = SET(destinationKey.ToArray(), newSetObject, ref objectContext);
+                }
+                else
+                {
+                    _ = EXPIRE(destinationKey, TimeSpan.Zero, out _, StoreType.Object, ExpireOption.None,
+                        ref lockableContext, ref objectContext);
+                }
+
+                return status;
+            }
+            finally
+            {
+                if (createTransaction)
+                    txnManager.Commit(true);
+            }
+        }
+
+        private GarnetStatus SortedSetUnion<TObjectContext>(ReadOnlySpan<ArgSlice> keys, ref TObjectContext objectContext,
+            out Dictionary<byte[], double> pairs, double[] weights = null, SortedSetAggregateType aggregateType = SortedSetAggregateType.Sum)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            pairs = default;
+
+            if (keys.Length == 0)
+                return GarnetStatus.OK;
+
+            // Get the first sorted set
+            var status = GET(keys[0].ToArray(), out var firstObj, ref objectContext);
+            if (status == GarnetStatus.OK)
+            {
+                if (firstObj.garnetObject is not SortedSetObject firstSortedSet)
+                {
+                    return GarnetStatus.WRONGTYPE;
+                }
+
+                // Initialize pairs with the first set
+                pairs = new Dictionary<byte[], double>(new ByteArrayComparer());
+                foreach (var (key, score) in firstSortedSet.Dictionary)
+                {
+                    pairs[key] = weights != null ? score * weights[0] : score;
+                }
+
+                // Process remaining sets
+                for (var i = 1; i < keys.Length; i++)
+                {
+                    status = GET(keys[i].ToArray(), out var nextObj, ref objectContext);
+                    if (status != GarnetStatus.OK)
+                        continue;
+
+                    if (nextObj.garnetObject is not SortedSetObject nextSortedSet)
+                    {
+                        pairs = default;
+                        return GarnetStatus.WRONGTYPE;
+                    }
+
+                    foreach (var (key, score) in nextSortedSet.Dictionary)
+                    {
+                        var weightedScore = weights != null ? score * weights[i] : score;
```

Comment:
Similar comment to SortedSetIntersection (SortedSetOps.cs line 989)

Reply:
It's already in the right format only